printer/PrintHandler.js

```diff
@@ -55,6 +55,30 @@ function determinePrinterForJob() {
   }
 }
 
+async function downloadFileFromS3(fileNo) {
+  const s3 = new AWS.S3({ apiVersion: '2012-11-05' });
+  const params = {
+    Bucket: PRINTING_BUCKET_NAME,
+    Key: `folder/${fileNo}.pdf`,
+  };
+  return new Promise((resolve) => {
+    try {
+      s3.getObject(params, function (err, dataFromS3) {
+        if (err) {
+          logger.error(`Unable to download file with id ${fileNo} :`, err);
+          resolve(false);
+        } else {
+          logger.info(`Successfully downloaded file with id ${fileNo}`);
+          resolve(dataFromS3);
+        }
+      });
+    } catch (e) {
+      logger.error('downloadFileFromS3 had an error:', e);
+      resolve(false);
+    }
+  });
+}
+ 
```

Comment:
explain what is going on in the rest of this code in some comments, its really hard to read when you dont know much about the file because you have no idea what is going on. I have no clue what is going on in this, just a few commnets would solve it